src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,180 @@
+                        .prepareStatement(queryForUpdateInCars)) {
+            deleteStmtCarsDrivers.setLong(1, car.getId());
+            updateStmtCars.setLong(1, car.getId());
+            deleteStmtCarsDrivers.executeUpdate();
+            updateStmtCars.executeUpdate();
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't update the data - "
+                    + car + "in table of cars or cars_drivers", exception);
+        }
+        create(car);
+        createCarsDriversDB(car);
         return car;
     }
 
     @Override
     public boolean delete(Long id) {
-        return Storage.getCars().removeIf(i -> i.getId().equals(id));
+        String queryForCars = "UPDATE cars SET car_deleted = true where car_id = ?;";
+        String queryForCarsDrivers = "delete from cars_drivers where car_id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statementForCars = connection.prepareStatement(queryForCars);
+                PreparedStatement statementForCarsDrivers = connection
+                        .prepareStatement(queryForCarsDrivers)) {
+            statementForCars.setLong(1, id);
+            statementForCarsDrivers.setLong(1, id);
+            return statementForCars.executeUpdate() > 0
+                    && statementForCarsDrivers.executeUpdate() > 0;
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't delete the data by id: " + id, exception);
+        }
     }
 
     public List<Car> getAllByDriver(Long driverId) {
-        return getAll().stream()
-                .filter(car -> car.getDrivers()
-                        .stream()
-                        .anyMatch(driver -> driver.getId().equals(driverId)))
-                .collect(Collectors.toList());
+        List<Car> carList = new ArrayList<>();
+        String query = "select distinct(c.car_id) from cars_drivers cd"
+                + " inner join cars c on cd.car_id = c.car_id "
+                + " where cd.driver_id = ? and c.car_deleted = false;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Optional<Car> optionalCar = get(resultSet.getObject(1, Long.class));
+                optionalCar.ifPresent(carList::add);
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't find drivers by id: " + driverId, exception);
+        }
+        return carList;
+    }
+
+    private Car getCarFromDB(ResultSet resultSet) throws SQLException {
+        ManufacturerDao manufacturerDao = new ManufacturerDaoImpl();
+        DriverDao driverDao = new DriverDaoImpl();
```

Comment:
let's not use these daos here
let's pass here the resultSet that contains info about the car itself and the manufacturer

and fill the car with drivers in an additional query in a separate private method

Reply:
done